lib/context2d.js

```diff
@@ -125,12 +125,11 @@ Object.defineProperty(Context2d.prototype, 'currentTransform', {
  */
 
 Context2d.prototype.__defineSetter__('fillStyle', function(val){
-  if (!val) return;
-  if ('CanvasGradient' == val.constructor.name
-    || 'CanvasPattern' == val.constructor.name) {
+  if (val instanceof CanvasGradient
+    || val instanceof CanvasPattern) {
```

Comment:
👍

Reply:
That would only work by coincidence I think. We've never tried to freeze those class structures.

Reply:
Cool 👍